packages/keyring/src/mnemonic.ts

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2018 - 2024 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { decrypt, encrypt } from '@alephium/shared-crypto'
+import { wordlist } from '@metamask/scure-bip39/dist/wordlists/english'
+
+export type MnemonicLength = 12 | 24
+
+export type EncryptedMnemonicVersion = 1 | 2
+
+export type DecryptMnemonicResult = {
+  decryptedMnemonic: Uint8Array
+  version: EncryptedMnemonicVersion
+}
+
+export class EncryptedMnemonicStoredAsUint8Array {
+  readonly version: EncryptedMnemonicVersion = 2
+  readonly mnemonic: Uint8Array
```

Comment:
Storing the mnemonic in local storage as a `Uint8Array` now, instead of a `string`.